grid.cpp

```diff
@@ -260,7 +262,27 @@ void initializeGrids(
       phiprof::start("Init moments");
       calculateInitialVelocityMoments(mpiGrid);
       phiprof::stop("Init moments");
- */
+      */
+
+      // Map Refinement Level to FsGrid
+      phiprof::start("Map Refinement Level to FsGrid");
+      const int *localDims = &momentsGrid.getLocalSize()[0];
+   
+      #pragma omp parallel for collapse(3)
+      for (int k=0; k<localDims[2]; k++) {
+         for (int j=0; j<localDims[1]; j++) {
+            for (int i=0; i<localDims[0]; i++) {
+
+               const std::array<int, 3> mapIndices = momentsGrid.getGlobalIndices(i,j,k);
+               const dccrg::Types<3>::indices_t  indices = {{(uint64_t)mapIndices[0],(uint64_t)mapIndices[1],(uint64_t)mapIndices[2]}}; //cast to avoid warnings
+               CellID dccrgCellID2 = mpiGrid.get_existing_cell(indices, 0, mpiGrid.mapping.get_maximum_refinement_level());
+               int amrLevel= mpiGrid.get_refinement_level(dccrgCellID2);
+               technicalGrid.get(i, j, k)-> RefLevel =amrLevel ;
```

Comment:
DCCRG's/mpiGrid's refinement level is 0 (or 1? darn, I don't remember) at coarsest, 1 (or 2?) for the next finer level and so on. I saw that you skip if the level is 0 in the smoothing loops, right? So I think that it is reversed.

Reply:
Guess resolved?